openslides_backend/action/actions/list_of_speakers/re_add_last.py

```diff
@@ -1,63 +1,83 @@
-from typing import Any, Dict, Iterable
+from typing import Any, Dict
 
-from ....models.models import ListOfSpeakers
+from ....models.models import ListOfSpeakers, Speaker
 from ....shared.exceptions import ActionException
-from ....shared.filters import FilterOperator
-from ....shared.interfaces.write_request_element import WriteRequestElement
-from ....shared.patterns import Collection, FullQualifiedId
-from ...action import Action
+from ....shared.filters import And, FilterOperator
+from ....shared.patterns import FullQualifiedId
+from ...generics.update import UpdateAction
 from ...util.default_schema import DefaultSchema
 from ...util.register import register_action
-from ..speaker.create_update_delete import SpeakerCreateAction
 
 
 @register_action("list_of_speakers.re_add_last")
-class ListOfSpeakersReAddLastAction(Action):
+class ListOfSpeakersReAddLastAction(UpdateAction):
     """
     Action to re-add the last speaker to the list.
     """
 
-    model = ListOfSpeakers()
+    model = Speaker()
     schema = DefaultSchema(ListOfSpeakers()).get_default_schema(
         required_properties=["id"],
         title="Re-add last speaker",
-        description="Adds the last speaker as new speaker.",
+        description="Moves the last speaker back to the top of the list.",
     )
 
-    def base_update_instance(self, instance: Dict[str, Any]) -> Dict[str, Any]:
+    def update_instance(self, instance: Dict[str, Any]) -> Dict[str, Any]:
+        # Check if list of speakers has speakers.
+        list_of_speakers_id = instance["id"]
         list_of_speakers = self.fetch_model(
```

Comment:
This fetch is unnecessary, you can just check if the filter call returned a result or not.